inc/rest-api/class-post-fields.php

```diff
@@ -204,21 +207,36 @@ public function register_meta_fields() {
 							);
 						}
 					} elseif ( 1 === $fm->limit && ! empty( $fm->children ) ) {
-						\FM_Gutenberg\register_meta_helper(
-							'post',
-							[ $post_type ],
-							$fm->name,
-							[
-								'type'         => 'object',
-								'show_in_rest' => [
-									'schema' => [
-										'type'       => 'object',
-										'properties' => $this->get_object_properties( $fm->children ),
-										'sanitize_callback' => $fm->sanitize,
+						if ( $fm->serialize_data ) {
+							\FM_Gutenberg\register_meta_helper(
+								'post',
+								[ $post_type ],
+								$fm->name,
+								[
+									'type'         => 'object',
+									'show_in_rest' => [
+										'schema' => [
+											'type'       => 'object',
+											'properties' => $this->get_object_properties( $fm->children ),
+											'sanitize_callback' => $fm->sanitize,
+										],
 									],
-								],
-							]
-						);
+								]
+							);
+						} else {
+							foreach ( $fm->children as $child ) {
+								\FM_Gutenberg\register_meta_helper(
+									'post',
+									[ $post_type ],
+									$fm->add_to_prefix ? $fm->name . '_' . $child->name : $child->name,
+									[
+										'default' => $child->default_value ?: '',
+										'sanitize_callback' => $child->sanitize,
+										'type'    => 'media' === $child->field_class ? 'integer' : 'string',
```

Comment:
Should `$child->field_class` have more strict type checking? are you looking to see if the property is set? or for a particular value?